### PR TITLE
test: add e2e test local

### PR DESCRIPTION
**Pull request by alegrey91**
## Overview
With this PR we are introducing a new **e2e** test suite for `host-scanner`.
The **e2e** tests will enable us to verify the right behavior of the provided endpoints (`/controlplaneinfo`, `/cniinfo`, ..).

In addition, we implemented also a **github-action** to automatically set up a test environment with **kind** and test 3 different **kubernetes** versions (currently `v1.25.0`, `v1.26.0`, `v1.27.0`)

## Additional Information
This PR provides also documentation about **e2e** tests overview and how to extend the tests for different providers.
The next step will be to extend the support to the most common cloud providers.

## Checklist before requesting a review

- [x] My code follows the style guidelines of this project
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .github/workflows/e2e-tests-local.yaml [hunk 1]
```diff
@@ -0,0 +1,110 @@
+name: e2e
+
+on:
+  push:
```
Comment (yuleib) [inline .github/workflows/e2e-tests-local.yaml:4]: consider of add pull_request also. We want to make sure during the dev. that e2e tests are passed

File: .github/workflows/e2e-tests-local.yaml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+#      - 'docs/**'
+#      - 'README.md'
+
+jobs:
+  kind:
+    name: Kubernetes
+    strategy:
+      fail-fast: false
+      matrix:
+        #k8s-version: ['v1.25.0', 'v1.26.0', 'v1.27.0']
```
Comment (yuleib) [inline .github/workflows/e2e-tests-local.yaml:21]: can we remove the comment ?

File: .github/workflows/e2e-tests-local.yaml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        k8s-version: ['v1.25.0']
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Clone host-scanner repository
+        uses: actions/checkout@v2
```
Comment (yuleib) [inline .github/workflows/e2e-tests-local.yaml:26]: replace checkout to v3 please. v2 is old

File: deployment/k8s-deployment.yaml [hunk 4]
```diff
@@ -1,60 +1,63 @@
...
-        operator: Exists
-        effect: NoSchedule
-      - key: node-role.kubernetes.io/master
-        operator: Exists
-        effect: NoSchedule
+      - operator: Exists
       containers:
       - name: host-sensor
-        image: quay.io/kubescape/host-scanner:latest
+        image: quay.io/kubescape/host-scanner:v1.0.57
```
Comment (yuleib) [inline deployment/k8s-deployment.yaml:35]: why we chnaged this to default ? we should keep using the latest while we are building HS. doesn't we  ?

File: .github/workflows/e2e-tests-local.yaml (hunk 1 above)
Comment (alegrey91) [reply .github/workflows/e2e-tests-local.yaml:4]: yes, `pull_request` event is currently commented to speed up local tests.
Once the PR will be ready, I'll remove the comments.

File: .github/workflows/e2e-tests-local.yaml (hunk 2 above)
Comment (alegrey91) [reply .github/workflows/e2e-tests-local.yaml:21]: same as above.

File: deployment/k8s-deployment.yaml (hunk 4 above)
Comment (alegrey91) [reply deployment/k8s-deployment.yaml:35]: yes, my fault. I uploaded the wrong file. Do we already have a tag `latest` for container image?

File: .github/workflows/e2e-tests-local.yaml (hunk 3 above)
Comment (alegrey91) [reply .github/workflows/e2e-tests-local.yaml:26]: fixed

File: deployment/host-scanner.yaml [hunk 5]
```diff
@@ -1,60 +1,69 @@
...
+      name: host-scanner
   template:
     metadata:
       labels:
-        name: host-sensor
+        name: host-scanner
     spec:
       tolerations:
-      # this toleration is to have the daemonset runnable on master nodes
+      # this toleration is to have the DaemonDet runnable on all nodes (including masters)
```
Comment (yuleib) [inline deployment/host-scanner.yaml:30]:
```suggestion
      # this toleration is to have the DaemonSet runnable on all nodes (including masters)
```

File: deployment/host-scanner.yaml [hunk 6]
```diff
@@ -1,60 +1,69 @@
...
       # remove it if your masters can't run pods
-      - key: node-role.kubernetes.io/control-plane
-        operator: Exists
-        effect: NoSchedule
-      - key: node-role.kubernetes.io/master
-        operator: Exists
-        effect: NoSchedule
+      - operator: Exists
       containers:
       - name: host-sensor
```
Comment (yuleib) [inline deployment/host-scanner.yaml:34]: aren't we changing it into host-scanner ?

File: deployment/host-scanner.yaml [hunk 7]
```diff
@@ -1,60 +1,69 @@
...
-        image: quay.io/kubescape/host-scanner:latest
+        image: kubescape/host-scanner:test
```
Comment (yuleib) [inline deployment/host-scanner.yaml:35]: how we are pulling the latest host-scanner now ? 
something is not clreared to me on this file now

File: deployment/host-scanner.yaml [hunk 8]
```diff
@@ -1,60 +1,69 @@
...
         ports:
-          - name: http
-            hostPort: 7888
+          - name: scanner # Do not change port name
             containerPort: 7888
+            protocol: TCP
         resources:
           limits:
-            cpu: 1m
+            cpu: 0.1m
```
Comment (yuleib) [inline deployment/host-scanner.yaml:48]: are we sure about this ? how did you test this performance ?

File: e2e/cloudproviderinfo_test.go [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+			resultBody := &sensor.CloudProviderInfo{}
+
+			resBody, err = ioutil.ReadAll(res.Body)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = json.Unmarshal(resBody, resultBody)
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(resultBody).To(Equal(cloudProviderInfo))
+		})
```
Comment (yuleib) [inline e2e/cloudproviderinfo_test.go:40]: need to fix and resolve those failures

File: e2e/cniinfo_test.go [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+			resultBody := &sensor.CNIInfo{}
+
+			resBody, err = ioutil.ReadAll(res.Body)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = json.Unmarshal(resBody, resultBody)
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(resultBody.CNINames).To(Equal(cniInfo.CNINames))
+
```
Comment (yuleib) [inline e2e/cniinfo_test.go:39]: need to fix and resolve those failures

File: e2e/cniinfo_test.go [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+			for i := range resultBody.CNIConfigFiles {
+				Expect(resultBody.CNIConfigFiles[i]).To(Equal(cniInfo.CNIConfigFiles[i]))
+			}
```
Comment (yuleib) [inline e2e/cniinfo_test.go:42]: need to fix and resolve those failures

**Review by yuleib — CHANGES_REQUESTED** [review]
@alegrey91 Well done with PR, 
i've put a few comments to resolve before merging, also let's review it together since it's a very bug PR with a lot of chnages

File: deployment/host-scanner.yaml (hunk 6 above)
Comment (alegrey91) [reply deployment/host-scanner.yaml:34]: I took it from [kubescape](https://github.com/kubescape/kubescape/blob/master/core/pkg/hostsensorutils/hostsensor.yaml#L32), so we can test `host-scanner` in the same way we deploy it on the cluster.

File: deployment/host-scanner.yaml (hunk 7 above)
Comment (alegrey91) [reply deployment/host-scanner.yaml:35]: In order to release the e2e local tests asap I left it in "legacy" mode (we basically build the image locally and then we load it into the cluster).
In the next few days, when the pipeline to test also managed clusters will be included, I'm going to unify the way we get the image. So the pipeline will build a `test` tag that will be pushed on quay.io and then all the clusters will get the image from here: `quay.io/kubescape/host-scanner:test`

File: deployment/host-scanner.yaml (hunk 8 above)
Comment (alegrey91) [reply deployment/host-scanner.yaml:48]: I took it from [kubescape](https://github.com/kubescape/kubescape/blob/master/core/pkg/hostsensorutils/hostsensor.yaml#L47), so we can test host-scanner in the same way we deploy it on the cluster.

**Pull request merged**